source/common/stats/thread_local_store.cc

```diff
@@ -61,6 +61,22 @@ std::list<GaugeSharedPtr> ThreadLocalStoreImpl::gauges() const {
   return ret;
 }
 
+std::list<HistogramSharedPtr> ThreadLocalStoreImpl::histograms() const {
+  // Handle de-dup due to overlapping scopes.
+  std::list<HistogramSharedPtr> ret;
+  std::unordered_set<std::string> names;
+  std::unique_lock<std::mutex> lock(lock_);
+  for (ScopeImpl* scope : scopes_) {
+    for (auto histogram : scope->central_cache_.histograms_) {
```

Comment:
The style-guide (https://google.github.io/styleguide/cppguide.html#auto) leaves some room for interpretation, but my reading is that in this case and a few others in this PR, you should use explicit types.

If you were using iterator objects, those are good candidates for 'auto' because their types are fairly obvious and not necessarily important to spell out.

e..g.
```
for (auto it = collection.begin(); it != collection.end(); ++it) {
  explicit_type& val = *it;
  ...
};
```
or
```
for (explicit_type& val : collection) {
  ...
};
```
But it's possible there's an established style here I'm going against the grain in suggesting.

Reply:
I am changing it like this
  for (ScopeImpl* scope : scopes_) {
    for (auto histogram : scope->central_cache_.histograms_) {
      const std::string& hist_name = histogram.first;
      const ParentHistogramSharedPtr& parent_hist = histogram.second;
      if (names.insert(hist_name).second) {
        ret.push_back(parent_hist);
      }
    }
  }